packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts

```diff
@@ -231,9 +239,17 @@ export async function deployToDesktop(
     windowsConfig && windowsConfig.project && windowsConfig.project.projectName
       ? windowsConfig.project.projectName
       : options.proj!;
-  const appPackageFolder = getAppPackage(options, projectName);
+  const windowsRoot =
```

Comment:
Is there a situation where windowsConfig doesn't have a folder/sourceDir?

If windowsConfig === null, then we need to error and exit before this point. Developers are allowed to set the value to null to block an platform from being detected.

Otherwise, it looks like windowsRoot is basically just the absolute path for sourceDir - and any logic for sourceDir should be calculated in config, for everyone's benefit, not here.

Reply:
I updated the code to fall back in this case to the slnFile. This should also take care of the case where folks pass -slnFile to provide a fallback from the config override.